components/eventing-controller/controllers/subscription/jetstream/reconciler.go

```diff
@@ -145,24 +145,25 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	// Check for valid sink
 	if err := r.sinkValidator.Validate(desiredSubscription); err != nil {
 		log.Errorw("sink URL validation failed", "error", err)
-		if syncErr := r.syncSubscriptionStatus(ctx, desiredSubscription, statusChanged, err); err != nil {
+		if syncErr := r.syncSubscriptionStatus(ctx, desiredSubscription, actualSubscription, statusChanged, err); err != nil {
 			return ctrl.Result{}, syncErr
 		}
 		// No point in reconciling as the sink is invalid, return latest error to requeue the reconciliation request
 		return ctrl.Result{}, err
 	}
 
 	// Synchronize Kyma subscription to JetStream backend
-	if syncErr := r.Backend.SyncSubscription(desiredSubscription); syncErr != nil {
+	syncErr := r.Backend.SyncSubscription(desiredSubscription)
+	if syncErr != nil {
```

Comment:
(nit)
```
if r.Backend.SyncSubscription(desiredSubscription) != nil {
```
instead of two lines.
```
syncErr := r.Backend.SyncSubscription(desiredSubscription)
if syncErr != nil {
```